### PR TITLE
chore: upgrade deno_lint, remove direct dprint dep

### LINKED ISSUE #6668: Multiple versions of swc_ecma_ast crate when updating deno to 1.1.3 via cargo
When updating deno to version 1.1.3 with cargo, the build of deno fails with errors such as:

```
error[E0053]: method `visit_import_decl` has an incompatible type for trait
    --> /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/deno-1.1.3/tsc.rs:1178:3
     |
1178 | /   fn visit_import_decl(
1179 | |     &mut self,
1180 | |     import_decl: &swc_ecma_ast::ImportDecl,
1181 | |     _parent: &dyn Node,
...    |
1188 | |     });
1189 | |   }
     | |___^ expected struct `swc_ecma_ast::module_decl::ImportDecl`, found a different struct `swc_ecma_ast::module_decl::ImportDecl`
     |
     = note: expected fn pointer `fn(&mut tsc::DependencyVisitor, &swc_ecma_ast::module_decl::ImportDecl, &(dyn swc_ecma_visit::Node + 'static))` (struct `swc_ecma_ast::module_decl::ImportDecl`)
                found fn pointer `fn(&mut tsc::DependencyVisitor, &swc_ecma_ast::module_decl::ImportDecl, &(dyn swc_ecma_visit::Node + 'static))` (struct `swc_ecma_ast::module_decl::ImportDecl`)
     = note: perhaps two different versions of crate `swc_ecma_ast` are being used?
```

### PR DESCRIPTION
**Pull request by bartlomieju**
This PR removes direct dependency on `dprint-plugin-typescript` and `swc_ecma_visit`. Both of these dependencies have been pushed out to `deno_lint` crate. This should make it a bit easier to do the upgrades and prevent having conflicting crate requirements.

Waiting on https://github.com/denoland/deno_lint/pull/220
Fixes #6668

### CODE REVIEW

**Comment by bartlomieju** [comment]
CC @dsherret

File: cli/fmt.rs [hunk 1]
```diff
@@ -9,10 +9,10 @@
 
 use crate::colors;
 use crate::diff::diff;
+use crate::dprint_plugin_typescript as dprint;
```
Comment (ry) [inline cli/fmt.rs:12]: where does it come from if not from a crate dependency? I'm confused.

Comment (dsherret) [reply cli/fmt.rs:12]: deno_lint re-exports it

Comment (ry) [reply cli/fmt.rs:12]: Does deno_lint actually use dprint_plugin_typescript? It doesn't seem like it.

**Pull request merged**